msim/src/sim/net/mod.rs

```diff
@@ -177,6 +182,23 @@ impl HostNetworkState {
             .ok_or_else(|| io::Error::new(io::ErrorKind::NotConnected, "no such socket"))?;
         Ok(cb(socket))
     }
+
+    fn delete_node(&mut self, id: NodeId) {
+        let to_remove: Vec<_> = self
+            .sockets
+            .iter()
+            .filter_map(|((node, fd), _)| {
+                if *node == id {
+                    Some((*node, *fd))
```

Comment:
style nit - feel free to ignore: why not just add these lines in the for loop and remove the `filter_map`?

Reply:
because there's no easy way to modify a container while iterating over it

Reply:
oh yes, my mistake :)